### PR TITLE
Add user store sql implementation

### PR DESCRIPTION
**Pull request by iomodo**
This PR adds sql implementation of the user store

### CODE REVIEW

File: core/kg/store/sqlstore/user_store.go [hunk 1]
```diff
@@ -39,25 +44,95 @@ func newSqlUserStore(sqlStore *SqlStore) store.UserStore {
 }
 
 func (us SqlUserStore) Save(user *model.User) (*model.User, error) {
-	return nil, nil
+	now := time.Now().Unix()
+
+	updatedUser := user.Clone()
```
Comment (giolekva) [inline core/kg/store/sqlstore/user_store.go:49]: If we always do a manual clone here why not take user mode.User as an argument (not reference, without *)?

File: core/kg/store/sqlstore/user_store.go [hunk 2]
```diff
@@ -39,25 +44,95 @@ func newSqlUserStore(sqlStore *SqlStore) store.UserStore {
...
+
+	if updatedUser.ID == "" {
+		updatedUser.ID = common.NewID()
+	}
+	updatedUser.CreateAt = now
```
Comment (giolekva) [inline core/kg/store/sqlstore/user_store.go:54]: Will Save be used for both creating new and updating existing user?
If so, CreateAt stay the same if it is an update.

File: core/kg/store/sqlstore/user_store.go [hunk 3]
```diff
@@ -39,25 +44,95 @@ func newSqlUserStore(sqlStore *SqlStore) store.UserStore {
...
+	updatedUser.UpdateAt = now
+	updatedUser.DeleteAt = 0
+	updatedUser.LastPasswordUpdate = now
```
Comment (giolekva) [inline core/kg/store/sqlstore/user_store.go:57]: Again if it is an update but password has not actually been changed then we should not touch LastPasswordUpdate field

File: core/kg/store/sqlstore/user_store.go [hunk 4]
```diff
@@ -39,25 +44,95 @@ func newSqlUserStore(sqlStore *SqlStore) store.UserStore {
...
 }
 
 func (us SqlUserStore) Count() (int64, error) {
-	return 0, nil
+	query := us.getQueryBuilder().
+		Select("count(*)").
+		From("Users").
+		Where(sq.Eq{"delete_at": 0})
+	row := query.QueryRow()
+
```
Comment (giolekva) [inline core/kg/store/sqlstore/user_store.go:88]: rm empty line here and bellow

**Review by giolekva — APPROVED** [review]
Approved after addressing these comments.

File: core/kg/store/sqlstore/user_store.go (hunk 2 above)
Comment (iomodo) [reply core/kg/store/sqlstore/user_store.go:54]: Will fix that when we add an `update` functionality, not sure if it should be in `save` or a totally different method. If in different method probably `save` is a bad name.

**Pull request closed without merging**